### PR TITLE
Set MySQL sink default isolation level to  read-committed

### LINKED ISSUE #3589: MySQL reports Deadlock when TiCDC replicating data
### What did you do?

In 1 TiDB, 1 TiKV, 1 TiCDC cluster, set up a changefeed that replicates all tables to downstream MySQL.

MySQL may reports deadlock error.

Reproduce:

* Extends TiDB GC lifetime to 2400h.
* Load data to TiDB, tiup bench tpcc prepare and tiup bench tpcc run
* Create a changefeed `cfa` whose start-ts = TiDB gc safepoint, and replicates data to MySQL.
* Wait checkpoint lag close to 10s.
* Remove changefeed `cfa` and create a changefeed `cfb` use the same config as `cfa`.

We can find deadlock errors in TiCDC log.

MySQL deadlock log: [deadlock.log](https://github.com/pingcap/ticdc/files/7586652/deadlock.log)

Deadlock table schema

```sql
CREATE TABLE `order_line` (
  `ol_o_id` int(11) NOT NULL,
  `ol_d_id` int(11) NOT NULL,
  `ol_w_id` int(11) NOT NULL,
  `ol_number` int(11) NOT NULL,
  `ol_i_id` int(11) NOT NULL,
  `ol_supply_w_id` int(11) DEFAULT NULL,
  `ol_delivery_d` datetime DEFAULT NULL,
  `ol_quantity` int(11) DEFAULT NULL,
  `ol_amount` decimal(6,2) DEFAULT NULL,
  `ol_dist_info` char(24) DEFAULT NULL,
  PRIMARY KEY (`ol_w_id`,`ol_d_id`,`ol_o_id`,`ol_number`)
) ENGINE=InnoDB DEFAULT CHARSET=latin1
```

### What did you expect to see?

No deadlock error.

### What did you see instead?

MySQL Deadlock error.

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
v5.2.1
```

TiCDC version (execute `cdc version`):

```console
v5.2.1
```

MySQL version

```
5.7.36
```

### PR DESCRIPTION
**Pull request by maxshuang**
### What problem does this PR solve?
Issue Number: close #3589 

### What is changed and how it works?
Add transaction_level="read-committed" to dsn

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update key monitor metrics in both TiCDC document and official document

### Release note 

```release-note
Set MySQL sink default isolation level to  read-committed to reduce `deadlock` in MySQL
```

### CODE REVIEW

**Review by amyangfei — APPROVED** [review]

File: cdc/sink/mysql_params.go [hunk 1]
```diff
@@ -46,6 +46,7 @@ const (
 	defaultWriteTimeout        = "2m"
 	defaultDialTimeout         = "2m"
 	defaultSafeMode            = true
+	defaultRCTxnIsolation      = "READ-COMMITTED"
```
Comment (zhaoxinyu) [inline cdc/sink/mysql_params.go:49]:
```suggestion
	defaultTxnIsolation      = "READ-COMMITTED"
```

Comment (maxshuang) [reply cdc/sink/mysql_params.go:49]: Besides, this variable may be changed by no way.

Comment (lance6716) [reply cdc/sink/mysql_params.go:49]: if so, I slightly prefer `defaultTxnIsolationRC`. rest lgtm

**Review by lance6716 — APPROVED** [review]

**Review by zhaoxinyu — APPROVED** [review]

**Pull request merged**